packages/framework/src/Model/AdvancedSearchOrder/Filter/OrderCustomerIdFilter.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\AdvancedSearchOrder\Filter;
+
+use Doctrine\ORM\QueryBuilder;
+use Shopsys\FrameworkBundle\Component\FlashMessage\FlashMessage;
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\AdvancedSearchFilterInterface;
+use Shopsys\FrameworkBundle\Model\Customer\CustomerFacade;
+use Shopsys\FrameworkBundle\Model\Customer\Exception\CustomerNotFoundException;
+use Symfony\Component\Form\Extension\Core\Type\NumberType;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+
+class OrderCustomerIdFilter implements AdvancedSearchFilterInterface
+{
+    public const string NAME = 'customerId';
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\CustomerFacade $customerFacade
+     * @param \Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface $flashBag
+     */
+    public function __construct(
+        protected readonly CustomerFacade $customerFacade,
+        protected readonly FlashBagInterface $flashBag,
+    ) {
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return self::NAME;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getAllowedOperators(): array
+    {
+        return [
+            self::OPERATOR_IS,
+            self::OPERATOR_NOT_SET,
+        ];
+    }
+
+    /**
+     * @return string
+     */
+    public function getValueFormType(): string
+    {
+        return NumberType::class;
+    }
+
+    /**
+     * @return array
+     */
+    public function getValueFormOptions(): array
+    {
+        return [];
+    }
+
+    /**
+     * @param \Doctrine\ORM\QueryBuilder $queryBuilder
+     * @param \Shopsys\FrameworkBundle\Model\AdvancedSearch\AdvancedSearchRuleData[] $rulesData
+     */
+    public function extendQueryBuilder(QueryBuilder $queryBuilder, $rulesData): void
+    {
+        foreach ($rulesData as $index => $ruleData) {
+            if ($ruleData->operator === self::OPERATOR_NOT_SET) {
+                $queryBuilder->andWhere('o.customer IS NULL');
+
+                continue;
+            }
+
+            try {
+                $customer = $this->customerFacade->getById((int)$ruleData->value);
+
+                $parameterName = 'customer_' . $index;
+                $queryBuilder->andWhere('o.customer = :' . $parameterName);
```

Comment:
:question: 

When filtering orders by multiple customers (ID 1, ID 2) it will not find any order. Wouldn't be better to collect all IDs and use `IN` clause (find all orders with ID 1 or ID 2, or ...). I do not know what is preferred filter, can you please check it ?